x.py

```diff
@@ -119,6 +119,11 @@ def build(dir: str, jobs: Optional[int], ghproxy: bool, ninja: bool, unittest: b
         cmake_options += ["-DCMAKE_C_COMPILER=clang", "-DCMAKE_CXX_COMPILER=clang++"]
     if D:
         cmake_options += [f"-D{o}" for o in D]
+
+    portable_flag_enabled = any("DPORTABLE" in o for o in cmake_options)
```

Comment:
would we require `portable_flag_enabled` be decided by `-DPORTABLE=1` or `-DPORTABLE=0`? Would

```
portable_flag_enabled = any("DPORTABLE" in o for o in cmake_options)
```

might cause `-DPORTABLE=1` disable portable?